src/docs/introduction/upgrading-to-stencil-three.md

```diff
@@ -0,0 +1,295 @@
+```diff
+  import { HTMLStencilElement, JSXBase } from "@stencil/core/internal";
+  import { NameType, Todo } from "./components/my-component/my-component";
++ export { NameType, Todo } from "./components/my-component/my-component";
+  export namespace Components {
+      interface MyComponent {
+        "first": NameType;
+      }
+  }
+  export interface MyComponentCustomEvent<T> extends CustomEvent<T> {
+      detail: T;
+      target: HTMLMyComponentElement;
+  }
+  declare global {
+      interface HTMLMyComponentElement extends Components.MyComponent, HTMLStencilElement {
+  }
+```
+This allows those types to be easily accessed from the root of the type distribution:
+```ts
+import { NameType, Todo } from '@my-lib/types';
+```
+
+When using `dist-custom-elements`, these types can now be accessed from the custom element output:
+```ts
+import { NameType, Todo } from '@my-custom-elements-output';
+```
+
+This _may_ clash with any manually created types in existing Stencil projects.
+Projects that manually create type definitions from `components.d.ts` will either need to:
+- remove the manually created type (if the types generated in `components.d.ts` suffice)
+- update their type creation logic to account for potential naming collisions with the newly generated types
+
+### Composition Event Handlers Renamed
+
+The names of Stencil's composition event handlers have been changed in order to
+correct a casing issue which prevented handlers from being called when events
+fired. The changes are as follows:
+
+| previous name                | new name                     |
+| ---------------------------- | ---------------------------- |
+| `onCompositionEnd`           | `onCompositionend`           |
+| `onCompositionEndCapture`    | `onCompositionendCapture`    |
+| `onCompositionStart`         | `onCompositionstart`         |
+| `onCompositionStartCapture`  | `onCompositionstartCapture`  |
+| `onCompositionUpdate`        | `onCompositionupdate`        |
+| `onCompositionUpdateCapture` | `onCompositionupdateCapture` |
+
+## Output Targets
+
+### `dist-custom-elements` Output Target
+#### Add `customElementsExportBehavior` to Control Export Behavior
+`customElementsExportBehavior` is a new configuration option for the output target.
+It allows users to configure the export behavior of components that are compiled using the output target.
+By default, this output target will behave exactly as it did in Stencil v2.0.0.
+
+This config option provides additional behaviors that will alter the default component export OR custom element definition behaviors
+for this target. The desired behavior can be set via the following in a project's Stencil config:
+
+```ts
+// stencil.config.ts
+import { Config } from '@stencil/core';
+export const config: Config = {
+  outputTargets: [
+    {
+      type: 'dist-custom-elements',
+      customElementsExportBehavior: 'default' | 'single-export-module',
```

Comment:
We're missing some options here:
```suggestion
      customElementsExportBehavior: 'default' | 'auto-define-custom-elements' | 'bundle' | 'single-export-module',
```

Reply:
Ah, good catch. That's on me, grabbed the docs from an earlier commit